### ISSUE TITLE
Windows genkey error

### ISSUE THREAD

**Issue by Crghilardi** [JuliaDocs/Documenter.jl#567]
Hello, 

I wanted to open up an issue following [this discussion about people in a windows env](https://discourse.julialang.org/t/documenter-jl-travis-genkeys-which-which-error-on-windows/5881/6).

I have made some progress but I stil have not gotten everything working correctly. 
I am not sure how much of this is able to be corrected in the codebase vs just adding to documentation outlining the process for windows more specifically. 
[I found this windows specific issue already open](https://github.com/JuliaDocs/Documenter.jl/issues/444) but I was not sure exactly how what I am running into fits with this. 

See the above linked Discourse post for steps taken so far:

So far I have 

1. Travis.genkeys() is not working 

Workaround: Generate keys manually from Gitshell using ssh-keygen and manually copy and paste strings to Travis and Github. 

Secondary problem: Travis requires escaping characters so I had to manually add backslashes to the string? Not sure if this is necessary? 
I also see .documenter.pub mentioned in various threads, do I need to create a file named that and move the physical .pub file anywhere?

I am now stuck on travis hanging on a passphrase on linux [as mentioned in this issue](https://github.com/JuliaDocs/Documenter.jl/issues/429) and completing the build successfully on Mac but not deploying the documentation. I have added the DOCUMENTER DEBUG keyword as well.

[Here](https://travis-ci.org/Crghilardi/ForestBiometrics.jl) is the link to my most recent attempt.

In conclusion:
What can I do to get around the passphrase error and get the documents building?

Is there any way to improve the windows genkey/deploy experience?
If not, I can put a more detailed write up together outlining what I had to do once I actually figure out how to get it working.

Thank you for your help

**Comment by mortenpi** [comment]
I had forgotten that we [`base64encode` the whole private key](https://github.com/JuliaDocs/Documenter.jl/blob/0fa59b1806e88776c1cb2c3d818fdb163443f1e7/src/Documenter.jl#L657) (and decode it when reading the variable on Travis). So this would have to be done by hand as well. No need to escape anything then. And I think this is related to the passphrase as well -- decoding a non-encoded key results in gibberish and `ssh` assumes that it's encrypted.

Improvement suggestions / PRs are certainly welcome. Having `Travis.genkeys` working on Windows would be best. Perhaps [adding stuff to the PATH](https://discourse.julialang.org/t/documenter-jl-travis-genkeys-which-which-error-on-windows/5881/4?u=mortenpi) might fix it. Otherwise, maybe `Travis.genkeys` can be split up, so that there would be a function people could use on Windows which would generate the copy-paste strings from manually generated private/public key files (basically [this part](https://github.com/JuliaDocs/Documenter.jl/blob/0fa59b1806e88776c1cb2c3d818fdb163443f1e7/src/Documenter.jl#L645-L659)).

The Windows-related issue you link is not related really -- that one is about about the differences in the output from Documenter, but not about UX.

**Comment by Crghilardi** [comment]
Hello, 

I wanted to give an update on what I have found so far. 
1. Adding the ~\Program Files\Git\usr\bin to my PATH did allow me to use the genkeys() function. This successfully generated a key that I could copy and paste into Travis/GitHub without having to modify anything. 
I also found that if a user has the GitHub desktop GUI version (rather than pure Git), the same cmd line functions are still installed, just hidden in a different folder. For me it looked like 
C:\Users\XXXX\AppData\Local\GitHub\PortableGit_f02737a78695063deace08e96d5042710d3e32db\usr\bin

2. I had to do some refactoring for my package in question, but I have finally had a successful Travis build that deployed to my gh-pages branch, so everything seems to be working mechanically now.
